### DISCUSSION TITLE
Unable to install 4.0.5 version

### DISCUSSION THREAD

**Discussion by shree0** [chevereto/chevereto#4] [Q&A]
Hello, 
I am trying to install chevereto 4.0.5 version using PHP 8.0 and PHP 8.1 version. But during 1st step of the installation, I got the `Something went wrong` error. I am not getting the exact error to debug it. 
I have created `env.php` inside` ...//app` folder and added the following:
<?php

return [
  'CHEVERETO_DB_NAME' => 'chevereto',
  'CHEVERETO_DB_PASS' => 'user_database_password',
  'CHEVERETO_DB_PORT' => '3306',
  'CHEVERETO_DB_TABLE_PREFIX' => 'chv_',
  'CHEVERETO_DB_USER' => 'user_database',
];

(replaced my database values with it). 

Please help how to resolve my issue.
Thanks.

**Comment by rodber** [comment]
Hi @shree0 

Refer to this message: https://chevereto.com/community/threads/re-something-went-wrong.14550/
(You need to debug to determine the actual error).

P.S. Happy new year!

**Comment by shree0** [comment]
Happy new year.

I have tried to install it in my Cpanel. After I create the `env.php` file I am getting the following errors. (If I put values to `env.php` I am getting a `Not Found` error.)

Errors in the Cpanel error log:
../index.php:12

.../app/legacy/entrypoints/index.php:25

.../app/src/Legacy/functions.php:979

..../app/vendor/chevere/chevere/src/Filesystem/FilePhpReturn.php:75

..../app/vendor/chevere/chevere/src/Filesystem/FilePhpReturn.php:70

..../app/vendor/chevere/chevere/src/Filesystem/FilePhpReturn.php:56

..../app/vendor/chevere/chevere/src/Filesystem/FilePhpReturn.php:126

..../app/env.php doesn't have any contents

Chevere\\Filesystem\\Exceptions\\FileWithoutContentsException thrown in .../app/vendor/chevere/chevere/src/Filesystem/FilePhpReturn.php:126

**Comment by rodber** [comment]
https://v4-docs.chevereto.com/guides/cpanel/

**Reply by shree0** [reply]
I followed the same steps to install but still, it was showing the same error `Something went wrong` and also in the error log I can also see the same errors. Didn't find a solution to resolve it.

**Comment by rodber** [comment]
Logs are there, is just that you aren't looking at the right place. As any system that depends on another, sometimes it gets harder to understand how that plays with logs.

There's a dirty hack to show debug:

At this line https://github.com/chevereto/chevereto/blob/f941723f875217e0c07ac03983eae6f877091194/app/legacy/load/register-handlers.php#L40 

Add this:

```php
$isDebug = true;
```

By doing that you will get the error always printed to the screen.

**Comment by rodber** [comment]
By the way, you can also define the env for error log device.

https://v4-docs.chevereto.com/application/configuration/environment.html#error-logging-variables

**Reply by shree0** [reply]
I have followed the steps you mentioned. Now I am not able to load the `.../install` URL. It is redirecting. (domain.com/chevereto/install is redirected to domain.com/install) which displays only a `Not Found` error.

I have copied the contents of `env-default.php` to `env.php`.  and replaced the database values in env.php.

**Comment by rodber** [comment]
When not using / you need to set the hostname path.

https://v4-docs.chevereto.com/application/configuration/environment.html#hostname-variables

**Reply by shree0** [reply]
I have set the hostname path as my domain and directory. (http://mydomain/mydirectory/) but my install page is not loading now showing a `This page isn’t working` error. Is my hostname path correct?

**Comment by shree0** [comment]
I have tried it in my cPanel also. Here I am getting the `Imagick` requirement error. 
In cPanel PHP 8.0 does not support  `Imagick`. extension

**Reply by shree0** [reply]
Can you please help with this?

**Comment by rodber** [comment]
We do have a cPanel guide.
https://v4-docs.chevereto.com/guides/cpanel/

Chevereto supports either GD or Imagick as image handling library.
https://v4-docs.chevereto.com/application/configuration/environment.html#image-handling-variables

**Reply by shree0** [reply]
I have replaced `'CHEVERETO_IMAGE_LIBRARY' => 'imagick',` to `'CHEVERETO_IMAGE_LIBRARY' => 'GD'. Now requirement error is not showing (mentioned above). But https:/.../../install URL is not loading, getting `This page isn’t working` error. I have enabled debug as you suggested earlier.

**Comment by rodber** [comment]
Check the server requirements, this needs virtual url support.

https://v4-docs.chevereto.com/application/stack/web-server.html
